mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICALToJsonAttribute.java

```diff
@@ -170,27 +169,38 @@ public void service(Mail mail) throws MessagingException {
     }
 
     private void setAttribute(Mail mail, Map<String, Calendar> calendars, Map<String, byte[]> rawCalendars) throws MessagingException {
-        Optional<String> sender = retrieveSender(mail);
+        Optional<MailAddress> sender = retrieveSender(mail);
         if (!sender.isPresent()) {
             LOGGER.info("Skipping {} because no sender and no from", mail.getName());
             return;
         }
+
+        MailAddress effectiveSender = sender.get();
+        MailAddress participant = AttributeUtils.getValueAndCastFromMail(mail, REPLY_TO_ATTRIBUTE, String.class)
```

Comment:
I don't get why you added this. Has it been required?

Reply:
As the explanation in the ticket,  there is an implementation from [Sabre VObject](https://github.com/sabre-io/vobject/blob/master/lib/ITip/Broker.php#L373-L382)  that add "sender" of the ITIP message to the list of event attendees if she is not in the list yet.

This logic is wrong in the case of the sender is not the attendee who changed her participant status. For example:
- When an attendee is an external user, then the sender to notice her change participant is OpenPaaS system (ex: no-reply@openpaas.org)

_Of course, I did not see any RFC for this, just check the google calendar for the behavior._

Reply:
Where does this REPLY_TO_ATTRIBUTE comes from? Aren't we suppose to read it from the message headers?